src/AsyncMqttClient.cpp

```diff
@@ -141,6 +158,25 @@ void AsyncMqttClient::_clear() {
 /* TCP */
 void AsyncMqttClient::_onConnect(AsyncClient* client) {
   (void)client;
+
+#if ASYNC_TCP_SSL_ENABLED
+  if (_secure && _secureServerFingerprints.size() > 0) {
+    bool sslFoundFingerprint = false;
+    SSL* clientSsl = _client.getSSL();
+    const uint8_t* fingerprint;
+    for (std::vector<std::array<uint8_t, SHA1_SIZE>>::iterator it = _secureServerFingerprints.begin(); it != _secureServerFingerprints.end(); ++it) {
+      fingerprint = it->data();
+      if (ssl_match_fingerprint(clientSsl, fingerprint) == SSL_OK) {
+        sslFoundFingerprint = true;
+      }
+    }
+    if (!sslFoundFingerprint) {
+      _client.close();
```

Comment:
Please replace this line with `_client.close(true)`. There's a reconnection problem without it.